src/v/storage/batch_cache.h

```diff
@@ -244,13 +253,16 @@ class batch_cache {
       , _is_reclaiming(o._is_reclaiming)
       , _size_bytes(o._size_bytes)
       , _reclaim_opts(o._reclaim_opts)
-      , _reclaim_size(_reclaim_opts.min_size) {
+      , _reclaim_size(_reclaim_opts.min_size)
+      , _background_reclaimer(std::move(o._background_reclaimer)) {
```

Comment:
is this move safe? it looks like when the `background_reclaimer` is moved it will contain a reference to the _old_ batch cache.

Reply:
great catch. I will make `batch_cache` to have deleted move constructor.